app/src/main/java/com/example/euriskocodechallenge/ui/home/viewmodel/MoreViewModel.kt

```diff
@@ -40,17 +31,30 @@ class MoreViewModel @Inject constructor(
     }
 
 
-    fun updateUser(fName: String, lName: String, imageBitmap: Bitmap) {
+    fun updateUser(fName: String, lName: String, imageString: String?) {
         viewModelScope.launch(Dispatchers.IO) {
             val userId = userDatabaseRepository.getLoggedInUserID()
-            userId.collect { id ->
-                userDatabaseRepository.getUserById(id).collectLatest { user ->
-                    user?.let {
-                        userDatabaseRepository.updateUser(it)
-                        UtilityFunctions.printLogs(Constants.TAG, "Updated User")
-                    }
+            userDatabaseRepository.getLoggedInUserID().collect {
+                var a: User? = null
+                var currentUser = userDatabaseRepository.getUserById(it)
+                currentUser.collect{ user ->
+                    a = user
                 }
+                a?.fName = fName
+                a?.lName = lName
+                a?.imageSrc = imageString
+
+                a?.let { updatedUser -> userDatabaseRepository.updateUser(updatedUser) }
             }
+
+//            userId.collect { id ->
```

Comment:
@martin371998 avoid pushing commented code, either add TODO: with description on why you're keeping it or remove if not needed anymore